frontend/src/views/StatblockEditorView.vue

```diff
@@ -686,23 +744,164 @@ export default defineComponent({
-			else if (cr == 0.25 && isIncrease) cr = 0.5;
-			else if (cr == 0.5 && isIncrease) cr = 1;
-			else if (cr == 0.125 && !isIncrease) cr = 0;
-			else if (cr == 0.25 && !isIncrease) cr = 0.125;
-			else if (cr == 0.5 && !isIncrease) cr = 0.25;
-			else if (cr == 1 && !isIncrease) cr = 0.5;
-			else {
-				if (isIncrease) cr = Math.min(30, cr + 1);
-				else cr = Math.max(0, cr - 1);
+		openCRCalculator(){
+			[this.crCalc.attackBonus, this.crCalc.dpr, this.crCalc.dc] = this.attackStats()
+
+			// Effective AC
+			this.crCalc.ac = this.calculateEffectiveAC()
+			this.crCalc.calculatedAC = this.crCalc.ac
+			
+			// Effective HP
+			this.crCalc.hp = this.calculateEffectiveHP()
+			this.crCalc.calculatedHP = this.crCalc.hp
+
+			this.crCalc.attackBonusCalc = this.crCalc.useDC == true ? this.crCalc.dc : this.crCalc.attackBonus
+			this.crCalc.expectedCR = this.data.description.cr
+			this.showCRModal = true
+		},
+		calculateEffectiveHP(): number{
+			let hp: number = this.data.defenses.hp.override ? this.data.defenses.hp.override : this.hpCalc()
+
+			// Adjust for vulnerabilities
+			if (this.data.defenses.vulnerabilities.length >0) hp = Math.floor(hp * this.crCalc.vulnerabilityModifier)
+
+			// Expected CR Multiplier			
+			hp = Math.floor(hp * this.expectedCRMultiplier())
+
+			return hp
+		},
+		expectedCRMultiplier(): number{
+			let multiplier = 1
+
+			if (this.data.defenses.immunities.length > 0){
+				if (this.crCalc.expectedCR >= 0 && this.crCalc.expectedCR <= 4){
+					multiplier = 2
+				} else if (this.crCalc.expectedCR <= 10){
+					multiplier = 2
+				} else if (this.crCalc.expectedCR <= 16){
+					multiplier = 1.5
+				} else {
+					multiplier = 1.25
+				}
+			} else if (this.data.defenses.resistances.length > 0){
+				if (this.crCalc.expectedCR >=0 && this.crCalc.expectedCR <=4){
+					multiplier = 2
+				} else if (this.crCalc.expectedCR <= 10){
+					multiplier = 1.5
+				} else if (this.crCalc.expectedCR <= 16){
+					multiplier = 1.25
+				}
+			}
+
+			return multiplier
+		},
+		calculateEffectiveAC(): number{
+			let ac: number = this.data.defenses.ac.ac
+			
+			ac += this.saveProficiencyModifier() + (this.crCalc.flies == true ? 2: 0)
+
+			return ac
+		},
+		saveProficiencyModifier(): number{
+			let numberProficientSaves = countProficientSaves(this.data.abilities.saves)
+			let saveModifier = 0
+
+			if (numberProficientSaves >=5){
+				saveModifier = 4
+			} else if (numberProficientSaves >= 3){
+				saveModifier = 2
 			}
 
-			this.data.description.cr = cr;
+			return saveModifier
+		},
+		calculateCR(): void{
+			let defenseCR: number = 0
+			let defenseRow: CRTableEntry = ChallengeRatingTable[defenseCR]
+			let offenseCR: number = 0
+			let offenseRow: CRTableEntry = ChallengeRatingTable[offenseCR]
+
+			// Ensure we stay within the bounds of the CR Table
+			const hp = this.crCalc.hp > 850 ? 850 : this.crCalc.hp
+			const dpr = this.crCalc.dpr > 320 ? 320 : this.crCalc.dpr
```

Comment:
These should use Math.min instead of a ternary. They should also pull the bounds from the table instead of set here as constants